tesla_http_proxy/DOCS.md

```diff
@@ -14,38 +14,103 @@ servers: nginx_proxy/*.conf
 
-Tesla will provide a Client ID and Client Secret.  Enter these in addon configuration and then Start it again.  Now the `regenerate auth` setting will be automatically disabled.
+Tesla will provide a Client ID and Client Secret. Enter these in addon configuration and then Start it again. Now the `regenerate auth` setting will be automatically disabled.
 
 Use the [Tesla Auth app](https://apps.apple.com/us/app/auth-app-for-tesla/id1552058613) to obtain a refresh token, by entering your Callback URL, Client ID, and Client Secret on the Fleet API page. This will open a webpage that asks for your Tesla account credentials.
 
 > [!TIP]
 > The first time you request a refresh token, it will also prompt to authorize your Client ID to access your Tesla account. Allow all scopes.
 
-Using the Home Assistant iOS app, open the Addon Web UI and click **Enroll public key in your vehicle**.  This should launch the Tesla app where it prompts for approval to "allow third-party access to your vehicle".  If you have multiple vehicles, you'll need to do this on each of them.
+Using the Home Assistant iOS app, open the Addon Web UI and click **Enroll public key in your vehicle**. This should launch the Tesla app where it prompts for approval to "allow third-party access to your vehicle". If you have multiple vehicles, you'll need to do this on each of them.
+
 > [!NOTE]
 > Your Tesla app must already be key-paired with the car.
 
 Configure the [Tesla integration](https://github.com/alandtse/tesla) to use this proxy. It should pre-fill the Client ID, URL, and certificate for you by reading them from this addon.
 
+## Specific setup guides
+
+### Cloudflare
+
+Setting up Tesla HTTP Proxy through cloudflare involves the following steps (all the steps are made with the available addons in home assistant, nothing external. Also it avoids the necessity to open ports on the router):
+
+1. Install **Cloudflared** addon. Then configure like this and start the addon.
+
+```yml
+external_hostname: ha.example.com
+additional_hosts:
+  - hostname: tesla.example.com
+    service: https://homeassistant-internal.example.com
+```
+
+2. Check on the **Cloudflare dashboard** that the two CNAME DNS Records appeared. If not read the **cloudflared** logs for errors.
+   ![image](https://github.com/llamafilm/tesla-http-proxy-addon/assets/1372028/42e637a8-4428-41c7-b244-099aa6844216)
+3. Create an `A` type record with name `homeassistant-internal` and IPv4 address pointing to the `LOCAL IP ADDRESS` of the machine hosting the homeassistant instance. Disable `proxied`.
+4. Install **Let's Encrypt** addon. (can be done with other addons, check discussions on the repo for alternatives). Then configure like this and make sure to configure the `cloudflare_api_token`. Make sure the `keyfile` and `certfile` aren't already been created. If so, delete them and proceed starting the addon. Read the logs to see when the files are created (takes more than 60 seconds).
```

Comment:
Might be worth adding the URL to the [Discussions](https://github.com/llamafilm/tesla-http-proxy-addon/discussions) for those who aren't quite github familiar.